src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py

```diff
@@ -848,18 +882,23 @@ def test_functionapp_create_registry_password_error(self, resource_group, storag
             'containerappmanagedenvironment', 40
         )
 
-        self.cmd('containerapp env create --name {} --resource-group {} --location {}'
-        .format(managed_environment_name, resource_group, WINDOWS_ASP_LOCATION_FUNCTIONAPP)).assert_with_checks([
+        existing_workspace_name = 'ExistingWorkspace'
+        workspace = self.cmd('monitor log-analytics workspace create -g {} -n {} -l francecentral'.format(resource_group, existing_workspace_name)).get_output_in_json()
+        shared_keys = self.cmd('monitor log-analytics workspace get-shared-keys -g {} -n {}'.format(resource_group, existing_workspace_name)).get_output_in_json()
+        self.cmd('containerapp env create --name {} --resource-group {} --location francecentral --logs-workspace-id {} --logs-workspace-key {}'
+        .format(managed_environment_name, resource_group, workspace['customerId'], shared_keys['primarySharedKey'])).assert_with_checks([
                      JMESPathCheck('name', managed_environment_name),
                      JMESPathCheck('resourceGroup', resource_group),
-                     JMESPathCheck('location', WINDOWS_ASP_LOCATION_FUNCTIONAPP)])
+                     JMESPathCheck('location', 'France Central')])
 
         with self.assertRaises(RequiredArgumentMissingError):   
             self.cmd('functionapp create -g {} -n {} -s {} --environment {} --runtime dotnet \
                     --functions-version 4 --image nginx  --registry-password 123'
                     .format(resource_group, functionapp_name, storage_account, managed_environment_name))
-            
-    @ResourceGroupPreparer(location=WINDOWS_ASP_LOCATION_FUNCTIONAPP)
+
+
+class FunctionAppManagedEnvironmentRegistryImage(LiveScenarioTest):
```

Comment:
Note: had to keep this test in live mode because it would fail otherwise as it would not find 'operationresults' polling matches in the recording but would always succeed in live mode.